# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: third_party/rust/wgpu-core/src/device/global.rs
# Commit: b3109ad14685
# Full Hash: b3109ad14685624ada7861083ff5332aec9a7818
# Author: Andy Leiserson <aleiserson@mozilla.com>
# Date: 2025-10-07 17:00 +0000
# Description:
#   Bug 1992842 - Update wgpu to upstream 1072b878 (2025-10-06) r=webgpu-reviewers,supply-chain-reviewers,ErichDonGubler
# ==============================================================================

diff -r 1d3f000cf1f9 -r b3109ad14685 third_party/rust/wgpu-core/src/device/global.rs
--- a/third_party/rust/wgpu-core/src/device/global.rs	Tue Oct 07 16:51:44 2025 +0000
+++ b/third_party/rust/wgpu-core/src/device/global.rs	Tue Oct 07 17:00:40 2025 +0000
@@ -1995,7 +1995,7 @@
 
                 let maintain_result;
                 (user_callbacks, maintain_result) =
-                    device.maintain(fence, wgt::PollType::Wait, snatch_guard);
+                    device.maintain(fence, wgt::PollType::wait_indefinitely(), snatch_guard);
 
                 match maintain_result {
                     // We're happy
@@ -2121,7 +2121,8 @@
 
             for (_id, device) in device_guard.iter() {
                 let poll_type = if force_wait {
-                    wgt::PollType::Wait
+                    // TODO(#8286): Should expose timeout to poll_all.
+                    wgt::PollType::wait_indefinitely()
                 } else {
                     wgt::PollType::Poll
                 };